peeker.go

```diff
@@ -71,6 +71,21 @@ func (p *peeker) ReadByte() (byte, error) {
 	return b, nil
 }
 
+func (p *peeker) ReadByteBuf(buf []byte) (byte, error) {
```

Comment:
If `peeker.ReadByte` is allocating `buf`, maybe just replace `peeker.lastByte` with a `[1]byte` and use that directly in `io.ReadFull`?

Reply:
ah, thats a good idea